### DISCUSSION TITLE
Wifi permanently drops after a couple hours

### DISCUSSION THREAD

**Discussion by reallydontbotherme** [pgale/lineage_15.1_Installation_TSV#5] [Q&A]
I'm hoping maybe its just a re-installation issue or something trivial in the settings or developer options but I'm encountering an issue where my wifi works fine upon booting and for a couple hours, but after that the wifi drops completely and requires a device restart for it to find wifi any wifi networks at all. Would this prompt a re-download and re-installation or is there an updated radio firmware I should look into getting? I'm having this across two different devices, so unfortunately its not just isolated to one drevice. 

Thank you for this fantastic guide, outside of these wifi snags which seem to sadly/positively be only isolated to myself the setup has been bulletproof!

**Comment by pgale** [comment]
What are you running on it? With Fully Kiosk, I'm not seeing that as it's keeping the Wi-Fi active. Probably something to do with Wi-Fi and the device sleeping or some peculiarity with your Wi-Fi setup maybe. My setups keep the Wi-Fi connected when the screen turns off and I don't use any of the system sleep modes.

**Reply by reallydontbotherme** [reply]
I'm just running it stock currently, so its probably just an issue with the device sleeping for a period of time. If the fix is to just run fullykiosk maybe I'll just resort to that. Thank you!

**Comment by pgale** [comment]
If using Fully Kiosk, the setting that might possibly help is `Prevent from Sleep while Screen Off` which prevents the device sleeping when the screen goes off (German grammar not so good though ;) )
I'm pretty sure I've had one of my Lineage 15.1 TSV's on for extended period without Fully Kiosk running but not concentrated on testing that. Might give it a go and see what happens.

**Comment by sco0ts** [comment]
I have the same issue.  I am actually just running the scrypted nvr app on the device.  After an hour (give or take),  screen turns black and the wifi shows not connected.  I actually need to pull the plug to get the wifi back.  No networks are seen when searching for ssids in the network settings.  Turning off/on wifi does not work either. 

Would love a way to keep this alive if that is the issue.  Maybe a tasker job to do something small?

Will try developer options to 'stay awake'

**Comment by nabahr** [comment]
Anybody have any updates on what's causing this? I just got a great deal on a bunch of the tsv devices and I've been setting them up to replace all the Google speakers using the view assist setup. So far I've got 3 devices running and I've seen this issue pop up on 2 of them. 
Both were running fine for up to a day or more before the wifi dropped out.
Using wallpanel, ip webcam, squeeze player all running full time. I have it set so that the screen doesn't sleep.

**Comment by entttom** [comment]
I pinned it on a 2.4 GHz Wi-Fi, and it works way better - didn't have a connection lost in the last 48 hours 
have you tried that? @nabahr @sco0ts

**Reply by vincenttor** [reply]
how do you force it to use 2.4 ghz, mine has dropouts all the time and i planned ot try this already.  i am using unifi accespoints but i can't force the tablet from there to go on 2.4 ghz, is it possible in the lineageos ?  i could not find this yet ,  thanks

**Reply by entttom** [reply]
I have a Unifi system and made a ssid with 2.4 ghz only and use this WiFi for the device (not exclusively - some other IOT devices too)

**Reply by vincenttor** [reply]
thanks, didnt thought of that, been looking in the controller of unifi but thats the solution yes,  only thing is it doesnt seem to help , also as someone says its the meshing/roaming maybe, i can tell u that its not the case, because when i forced it to my mobile hotspot on 2.4 ghz, it did  seem to respond better online ( browsing)  i t was not the solution because it still crashed the wifi connection and after a couple of times airplane mode or adapter off/on the tablet crashes,  its is to slow feels like it and done. 


Also noticed today that it has been online all night, checked the log of my network and it seemed online ,until now , came in the room and a sensor turned the screen on, started working on it and poof, magically wifi connection broke when it gets to much ( kiosk browser on homeassist page, restarted wifi and resulted in a black screen thus powercycle)  this unit i am going to put on android see how that goes and otherwise break it open see whats in side haha ( qurious whats in the stand)

**Comment by variasdesign** [comment]
I'm having the same issue with my unit, sadly. Using a 2.4 GHz network and the default launcher.

**Reply by nabahr** [reply]
I still don't know the root cause. I am using 6 TSV in total now, all set up the same way, and only one of them has the issue with the wifi dropping off. At least as far as I know, 4 of them are in bedrooms and I've told people that if there is any issue to just power cycle it, so it might be happening more than I know.

Anyway, the one I know has the issue. I've set up a tasker automation for when the wifi connection disconnects, then turn on airplane mode and then turn it back off after a few seconds. This fixes the wifi issue and auto connects to the wifi again. This solution has been working pretty consistently. I still need to power cycle them sometimes because the screen goes black and I have no way to wake it up again, even though I've tried every setting I can find to not turn off the screen.

**Reply by vincenttor** [reply]
thanks for the very fast replie,  this happens here as well.
i have 2 more laying here but i'm afraid these would do the same , but i could be lucky.

Hoping to force it but i can't find a way,  i did force the tablet on 2.4 ghz by my phone hotspot but in unifi i can't force it per user :( 
if i put the whole wifi network on 2.4ghz then its a waste of my 1Gbit fiber connection.

Thanks for the tasker tip, did not think of that right away when stop turn on.
Did found out quite fast that the turn wifi off and then on again works  but the things are sort of fragile when u push to many buttons or to fast because it does indeed go on black screen and won't pop on again.

Wel tomorrow i will try to install the other 2 and hope these wil not give to much problems otherwise it has been a waste of money , but this hasn't been the first time i have bought stuff for home automation that did not work out like i hoped it would haha.

Thanks , if ill find out something some day ill surely let you know here.

**Reply by sco0ts** [reply]
Please report back if you have the same issue on unit number 2. Also I think there may be some different hardware over time so maybe different depending on that.

**Reply by vincenttor** [reply]
btw, i forgot to write here but, the other units didnt gave the problem,  wrote on another reply here how and what, still need to reflash it and hope it works , otherwise spareparts

**Reply by DudeInMyrtleBeach** [reply]
> Hoping to force it but i can't find a way, i did force the tablet on 2.4 ghz by my phone hotspot but in unifi i can't force it per user :( if i put the whole wifi network on 2.4ghz then its a waste of my 1Gbit fiber connection.
> 

You can create multiple Wifi networks (SSID's) in the Unifi app on the same hardware and limit each to whatever frequencies you want.

**Comment by psgcooldog** [comment]
I have this same problem.  I think its related to the mesh networking.
Doesnt matter what band i use.

**Reply by vincenttor** [reply]
probably not, ive forced it on my mobile hotspot 2.4ghz.

While this looked better at the beginning  , browsing faster and more stable,  after some presses on the screen and it started to get difficult the adapter sort of crashed.
Also could not find any wifi network then anymore untill i restarted the adapter then it came online after a little while.

For me it feels like its to heavy for the tablets since when i used it and i noticed the screen sort of lagging the connection dropped few seconds after that.

Done another power cycle since the screen goes black sometimes when restarting the wifi adapter as well,  this unit seems not usable for me, but going to try and install the android os online see if that works better , if not then its a waste but what can u do...  don't know if its possible to put a usb-c lan adapter in it that it will be a lan connection by wire.

Probably won't work because of driver problem

**Reply by DudeInMyrtleBeach** [reply]
I have one of these using Fully Kiosk. It totally hangs - screen off and disconnected wifi - after a day or so, and it takes a power cycle to get it going again. One thing I noticed in my Fully Kiosk/HA integration is that the device always shows as 'unplugged' in the Integration.. Even when its running, it never shows 'plugged in'. The 'plugged-in/unplugged' sensor shows random wifi disconnects and after a period of time, it's just gone. I've attached a pic of this behavior. Maybe Fully Kiosk isn't keeping it alive as we all think? I'm also using Unifi... I wonder if that the thing that we all have in common? I might just grab some other wifi tech out of my closet and test this out.. Can't hurt.

Anyhoo, if I can't get this to work somewhat normally, I'll install it on a smart plug and have home assistant power cycle it when it drops for a period of time. Kind of a kludge, but I don't know what else to do.
<img width="472" alt="Lenovo" src="https://github.com/user-attachments/assets/bf161ad4-ec2f-4d2a-86a0-75a8dd9e5e21" />

**Reply by DudeInMyrtleBeach** [reply]
I disabled everything related to sleep and power savings/management in fully kiosk on mine, and it hasn't disconnected since I posted this 3 weeks ago.

**Comment by sco0ts** [comment]
I’m not using home assistant app with mine. Using scrypted to view my security cameras. So I don’t think HA is the issue

**Comment by vincenttor** [comment]
I have installed the other 2 tablets , no problems, i noticed it right away when booting and asing for wifi.
That first tablet that gave wifi problems was already searching quite long before it could find an accespoint/ssid.

Also no power cycles needed anymore they run pretty good , yes they are slower in response as i thought but they stand niceley and with kiosk they are quite usable .

The unit that seems "broken"  i will reinstall again, see if that solves anything, if not , its a faulty unit and for spare parts, if there are any.
 (doesnt matter 2.4ghz or 5ghz , they both work)

**Comment by matthewvinton** [comment]
I'm 90%+ sure this is a hardware issue. I had two of these units, one had the issue (wifi disconnects followed eventually by an unresponsive device) and one didn't. I reloaded the trouble one from scratch making very careful note of the process, and had the same issue. I decided that it was worth rolling the dice and trying a new device, and used the same build process on the new device. It has performed flawlessly for a couple months.

**Reply by pgale** [reply]
Not convinced. One of my units has been operating without any disconnects for a very long time - since I got them over a year ago. All of a sudden, it started disconnecting within around 12 hours of a restart (no pattern I can see though). The only thing I can see that changed apart from the HA versions is that I installed an additional UniFi mesh AP at around the same time. However, I've tried 2.4 and 5GHz networks, turned off all advanced settings, setup an IoT network with IoT compatibility (in UniFi) and loads of other stuff - nothing has helped so I'm at a bit of a loss. I have a spare unit so will install and swap that out and rebuild the problem unit to see if I get anywhere.

**Reply by sco0ts** [reply]
> Not convinced. One of my units has been operating without any disconnects for a very long time - since I got them over a year ago. All of a sudden, it started disconnecting within around 12 hours of a restart (no pattern I can see though). The only thing I can see that changed apart from the HA versions is that I installed an additional UniFi mesh AP at around the same time. However, I've tried 2.4 and 5GHz networks, turned off all advanced settings, setup an IoT network with IoT compatibility (in UniFi) and loads of other stuff - nothing has helped so I'm at a bit of a loss. I have a spare unit so will install and swap that out and rebuild the problem unit to see if I get anywhere.

I also have a unifi system....I am currently at max number of wifi networks so i cant make a test network.  Would you mind trying to isolate that wifi network to only one AP?  I wonder if the issue is the unit trying to roam between APs?

**Reply by pgale** [reply]
I had the TSV locked to one AP so probably not that unfortunately.

**Comment by vincenttor** [comment]
I noticed the same on my other 2  devices,  I didn’t have the time yet to reinstall the faulty one , hopefully this decides to work as well in the end.  Will let you guys know when I get to it but it will probably one for parts I’m afraid of.

**Comment by silentdefault** [comment]
I don’t read all the comments, but some time ago, I tried using the image and noticed that the Wi-Fi was continuously disconnecting, giving me only a few seconds of internet at a time. In recent days, my roommate got a "new" phone and experienced the same issues. I found out that my Wi-Fi was the problem.

I have an Omada controller with TP-Link APs. I tried changing some settings and discovered that filtering certain protocols helped prevent this from happening. I haven't fully tested all the configurations or figured out exactly which settings are necessary, but this is the main change I made, and it worked.

![image](https://github.com/user-attachments/assets/38b5de68-84fe-4083-8b0f-430554e980bb)

**Reply by autumnwalker** [reply]
Interesting. I have the same wireless dropping and I'm on an Omada network. I've tried turning on "others" for now to see what that does.

**Reply by autumnwalker** [reply]
"others" alone did not solve the problem. I've now enabled ND and IGMP ... hoping I can keep mDNS on.

**Reply by autumnwalker** [reply]
Unfortunately this did not work for me.

**Comment by reallydontbotherme** [comment]
I'm not sure if this will help everyone, but as the OP I actually sorted this out with an app that might be placebo but might also be doing something to keep the device alive and on the wifi.

I like having my HA screens turn off and only activate when I am nearby, so I installed an android app called YAKK: https://play.google.com/store/apps/details?id=bk.appz.yakk&hl=en_US

Basically it enables the camera to be a motion sensor that turns the screen on when its off and it detects enough motion IE me walking near the device. I assume it requires keeping the device active 24/7 to actually do the monitoring just with the screen turned off, but after I installed this on both devices, my wifi issues went away. 

Completely anecdotal, maybe it will help you maybe it will not. I do like the app though paired with this device because I can also "disable" a screen by just sliding the camera shutter shut, such as the one at my workspace I only want turned on when I am working.

Good luck and godspeed!

**Reply by pgale** [reply]
The unit I suddenly started having a problem with is on 24/7 now and the dashboard has a CCTV camera feed running continually, so that should keep the WiFi alive. However it doesn't. I installed a spare to replace it and so far, that unit is behaving. Very odd.

**Comment by autumnwalker** [comment]
Some more data. I set up the TSV to logcat to the device so I could try and capture the disconnect. Here's what I'm seeing (.30.1 is my gateway):

> 09-02 23:26:33.405397  1000  2390 W IpReachabilityMonitor: ALERT: NeighborEvent{elapsedMs=54801990, 192.168.30.1, [(null)], RTM_NEWNEIGH, NUD_FAILED}
> 09-02 23:26:33.410686  1000  2390 W IpReachabilityMonitor: FAILURE: LOST_PROVISIONING, NeighborEvent{elapsedMs=54801990, 192.168.30.1, [(null)], RTM_NEWNEIGH, NUD_FAILED}
> 09-02 23:26:33.445837  1514  1514 I wpa_supplicant: wlan0: CTRL-EVENT-DISCONNECTED bssid=[MAC ADDRESS] reason=3 locally_generated=1
> 09-02 23:26:33.451695  1514  1514 E wpa_supplicant: nl80211: Failed to open /proc/sys/net/ipv4/conf/wlan0/drop_unicast_in_l2_multicast: No such file or directory
> 09-02 23:26:33.451755  1514  1514 E wpa_supplicant: nl80211: Failed to set IPv4 unicast in multicast filter
> 09-02 23:26:33.452490  1514  1514 E wpa_supplicant: nl80211: Failed to open /proc/sys/net/ipv4/conf/wlan0/drop_unicast_in_l2_multicast: No such file or directory
> 09-02 23:26:33.452533  1514  1514 E wpa_supplicant: nl80211: Failed to set IPv4 unicast in multicast filter
> 09-02 23:26:33.453036  1000  1100 D WifiStateMachine: WifiStateMachine: Leaving Connected state
> 09-02 23:26:33.454227  1000  1102 D DhcpClient: doQuit
> 09-02 23:26:33.455208  1000  2400 D DhcpClient: Receive thread stopped
> 09-02 23:26:33.456670  1000  1100 E WifiVendorHal: stopRssiMonitoring(l.2156) failed {.code = ERROR_NOT_AVAILABLE, .description = }
> 09-02 23:26:33.456763  1000  1100 W WifiConfigManager: Looking up network with invalid networkId -1
> 09-02 23:26:33.457078  1514  1514 E wpa_supplicant: wpa_driver_nl80211_driver_cmd: failed to issue private commands
> 09-02 23:26:33.457215  1000  1100 E SupplicantStaIfaceHal: ISupplicantStaIface.setSuspendModeEnabled failed: FAILURE_UNKNOWN, 
> 09-02 23:26:33.457711  1514  1514 E wpa_supplicant: wpa_driver_nl80211_driver_cmd: failed to issue private commands
> 09-02 23:26:33.457827  1000  1100 E SupplicantStaIfaceHal: ISupplicantStaIface.setBtCoexistenceMode failed: FAILURE_UNKNOWN, 

I don't understand what this means, but it appears to drop the wifi connection for some reason and then it has a bunch of problems trying to open some file and then it it looking for an "invalid networkid -1".

**Comment by autumnwalker** [comment]
I find this really hard to believe, but disabling bluetooth seems to have resolved the wifi drops for me. I ran my logcat through AI and it noticed issues with the BT stack. I disabled bluetooth in `settings > connected devices > bluetooth` and for the past 5 days I have had zero wifi drops / device lockups.

**Reply by pgale** [reply]
Interesting - not too hard to believe. I rarely get drops on my devices but will try that and see what happens longer term.

**Reply by dimatx** [reply]
Datapoint... For what it's worth, one of my TSVs dropped from WiFi despite having Bluetooth off.

<img width="1102" height="212" alt="image" src="https://github.com/user-attachments/assets/853d9537-3074-49c1-80e0-de253e3e5253" />

**Reply by matthewvinton** [reply]
@autumnwalker I want to thank you for the suggestion to disable Bluetooth. I suspect that wifi stability issues may have a variety of causes, but that suggestion at least for me has been night-and-day. It has allowed me to utilize my problem device, which after disabling Bluetooth is as stable as my other two devices. Thanks again.
